rgw/v2/tests/s3_swift/reusable.py

```diff
@@ -1424,3 +1424,83 @@ def bucket_reshard_manual(bucket, config):
         log.info(f"num_shards for bucket {bucket.name} after reshard are {shards}")
     else:
         raise TestExecError(f"Bucket {bucket.name} not resharded to {config.shards}")
+    verify_attrs_after_resharding(bucket)
+
+
+def set_dynamic_reshard_ceph_conf(config, ssh_con):
+    ceph_conf = CephConfOp(ssh_con)
+    log.info("sharding type is dynamic. setting ceph conf parameters")
+    time.sleep(15)
+    log.info("making changes to ceph.conf")
+    ceph_conf.set_to_ceph_conf(
+        "global",
+        ConfigOpts.rgw_max_objs_per_shard,
+        str(config.max_objects_per_shard),
+        ssh_con,
+    )
+
+    ceph_conf.set_to_ceph_conf(
+        "global", ConfigOpts.rgw_dynamic_resharding, "True", ssh_con
+    )
+    ceph_conf.set_to_ceph_conf(
+        "global",
+        ConfigOpts.rgw_max_dynamic_shards,
+        str(config.max_rgw_dynamic_shards),
+        ssh_con,
+    )
+
+    ceph_conf.set_to_ceph_conf(
+        "global",
+        ConfigOpts.rgw_reshard_thread_interval,
+        str(config.rgw_reshard_thread_interval),
+        ssh_con,
+    )
+
+
+def bucket_reshard_dynamic(bucket, config):
+    # for dynamic,
+    # the number of shards  should be greater than   [ (no of objects)/(max objects per shard) ]
+    # example: objects = 500 ; max object per shard = 10
+    # then no of shards should be at least 50 or more
+    resharding_sleep_time = config.rgw_reshard_thread_interval
+    log.info(
+        "verification of dynamic resharding starts after waiting for reshard_thread_interval:"
+        + f"{resharding_sleep_time} seconds"
+    )
+    time.sleep(resharding_sleep_time)
+
+    num_shards_expected = config.objects_count / config.max_objects_per_shard
+    log.info("num_shards_expected: %s" % num_shards_expected)
+    op = utils.exec_shell_cmd("radosgw-admin bucket stats --bucket %s" % bucket.name)
+    json_doc = json.loads(op)
+    bucket_id = json_doc["id"]
+    num_shards_created = json_doc["num_shards"]
+    log.info("no_of_shards_created: %s" % num_shards_created)
+    log.info("Verify if resharding list is empty")
+    reshard_list_op = json.loads(utils.exec_shell_cmd("radosgw-admin reshard list"))
+    if not reshard_list_op:
+        log.info(
+            "for dynamic number of shards created should be greater than or equal to number of expected shards"
+        )
+        log.info("no_of_shards_expected: %s" % num_shards_expected)
+        if int(num_shards_created) >= int(num_shards_expected):
```

Comment:
If num_shards_created < num_shards_expected we need to raise assertion ryt?

Reply:
@ckulal , addressed the comment